qt/main.cpp

```diff
@@ -169,11 +173,7 @@ int main(int argc, char * argv[])
 #if defined(OMIM_OS_MAC)
     apiOpenGLES3 = app.arguments().contains("es3", Qt::CaseInsensitive);
 #elif defined(OMIM_OS_LINUX)
-    // TODO: Implement proper runtime version detection in a separate commit
-    // Currently on Linux in a maximum ES2 scenario,
-    // the GL function pointers wouldn't be properly resolved anyway,
-    // so here at least a possibly successful path is chosen.
-    apiOpenGLES3 = true;
+    apiOpenGLES3 = !FLAGS_es2;
```

Comment:
Shouldn't the available es2 or es3 version be detected in runtime automatically?

Reply:
Ideally yes. And I think that should be a topic of a separate PR, because this PR is already very old. 
Do you think it is acceptable to do that in a separate PR?

Reply:
It will probably prolong the review process, but I have addedd code for that anyway.